src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java

```diff
@@ -1678,4 +1692,110 @@ public void handleSettingsAdded(SettingsInfo settings) {
+            if (wmsStore.getWorkspace() != null) {
+                wmsStore.setWorkspace(
+                        getById(wmsStore.getWorkspace().getId(), WorkspaceInfo.class));
+            }
+        }
+
+        @Override
+        public void visit(FeatureTypeInfo featureType) {
+            if (featureType.getNamespace() != null) {
+                featureType.setNamespace(
+                        getById(featureType.getNamespace().getId(), NamespaceInfo.class));
+            }
+            featureType.setStore(getById(featureType.getStore().getId(), StoreInfo.class));
+        }
+
+        @Override
+        public void visit(CoverageInfo coverage) {
+            if (coverage.getNamespace() != null) {
+                coverage.setNamespace(
+                        getById(coverage.getNamespace().getId(), NamespaceInfo.class));
+            }
+            coverage.setStore(getById(coverage.getStore().getId(), StoreInfo.class));
+        }
+
+        @Override
+        public void visit(LayerInfo layer) {
+            if (layer.getDefaultStyle() != null) {
+                layer.setDefaultStyle(getById(layer.getDefaultStyle().getId(), StyleInfo.class));
+            }
+            Set<StyleInfo> newStyles = new HashSet<>();
+            for (StyleInfo style : layer.getStyles()) {
+                if (style != null) {
+                    newStyles.add(getById(style.getId(), StyleInfo.class));
+                }
+            }
+            layer.getStyles().clear();
+            layer.getStyles().addAll(newStyles);
+        }
+
+        @Override
+        public void visit(StyleInfo style) {
+            if (style.getWorkspace() != null) {
+                style.setWorkspace(getById(style.getWorkspace().getId(), WorkspaceInfo.class));
+            }
+        }
+
+        @Override
+        public void visit(LayerGroupInfo layerGroup) {
+            if (layerGroup.getWorkspace() != null) {
+                layerGroup.setWorkspace(
+                        getById(layerGroup.getWorkspace().getId(), WorkspaceInfo.class));
+            }
+            for (int i = 0; i < layerGroup.getLayers().size(); i++) {
+                if (layerGroup.getLayers().get(i) != null) {
+                    layerGroup
+                            .getLayers()
+                            .set(
+                                    i,
+                                    getById(
+                                            layerGroup.getLayers().get(i).getId(),
+                                            PublishedInfo.class));
+                }
+            }
+        }
+
+        @Override
+        public void visit(WMSLayerInfo wmsLayer) {
+            if (wmsLayer.getNamespace() != null) {
+                wmsLayer.setNamespace(
+                        getById(wmsLayer.getNamespace().getId(), NamespaceInfo.class));
+            }
+            wmsLayer.setStore(getById(wmsLayer.getStore().getId(), StoreInfo.class));
+        }
+    }
```

Comment:
It probably wouldn't hurt to override the default implementations of visit(WMTSStoreInfo store) and visit(WMTSLayerInfo layer).

Reply:
why, they don't seem to have any references?

Reply:
Those two methods were added using Java 8 default methods.

Reply:
I understand, but I see no reason to override them?

Reply:
Why would you need to do something for WMSStoreInfo and WMSLayerInfo but not WMTSStoreInfo and WMTSLayerInfo?

Reply:
sorry, you are right, it does ahve references! I changed it